src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -260,14 +286,14 @@ internal class ModelCheckingStrategy(
     ) {
         private var lastNotInitializedNode: SwitchChoosingNode? = initialLastNotInitializedNode
         private lateinit var interleavingFinishingRandom: Random
-        private lateinit var nextThreadToSwitch: Iterator<Int>
+        private var currentInterleavingPosition = 0
         private var lastNotInitializedNodeChoices: MutableList<Choice>? = null
         private var executionPosition: Int = 0
 
         fun initialize() {
             executionPosition = -1 // the first execution position will be zero
             interleavingFinishingRandom = Random(2) // random with a constant seed
-            nextThreadToSwitch = threadSwitchChoices.iterator()
+            currentInterleavingPosition = 0
```

Comment:
Could we initialize it as:
```kotlin
// set to -1 if loop detector controls thread switches
currentInterleavingPosition = if (!loopDetector.replayModeEnabled) 0 else -1 
```

Then it will also simplify the checked invariant later in `chooseThread`.